components/style/matching.rs

```diff
@@ -453,6 +453,15 @@ trait PrivateMatchMethods: TElement {
                                        &new_values,
                                        None);
 
+            // If computed values for custom properties changed, we should cascade these changes to
+            // children (custom properties are all inherited).
+            // https://www.w3.org/TR/css-variables/#defining-variables
+            // (With Properties & Values, not all custom properties will be inherited!)
+            if old_values.as_ref().map(|v| v.as_ref()).and_then(|v| v.custom_properties()) !=
+               new_values.custom_properties() {
+                child_cascade_requirement = ChildCascadeRequirement::MustCascadeDescendants
```

Comment:
In particular, you need to set `StyleChange::Changed` even if `damage.is_empty()`, in the case custom properties have changed.